pkg/userop/client.go

```diff
@@ -376,96 +372,67 @@ func (c *backend) SendUserOp(ctx context.Context, op UserOperation) (<-chan Rece
 	return done, nil
 }
 
-// waitForUserOpEvent waits for a user operation to be committed on block.
-func waitForUserOpEvent(
+// subscribeUserOpEvent waits for a user operation to be committed on block.
+func subscribeUserOpEvent(
 	ctx context.Context,
-	pollPeriod time.Duration,
 	cancel context.CancelFunc,
 	client EthBackend,
 	done chan<- Receipt,
 	entryPoint common.Address,
 	userOpHash common.Hash,
 ) {
-	ticker := time.NewTicker(pollPeriod * time.Millisecond)
-	defer ticker.Stop()
-	defer close(done)
 
-	fromBlock, err := client.BlockNumber(ctx)
+	entryPointContract, err := entry_point_v0_6_0.NewEntryPoint(entryPoint, client)
 	if err != nil {
-		slog.Error("failed to get block number", "error", err)
+		slog.Error("failed to connect to the entry point contract", "error", err)
 		cancel()
```

Comment:
What happens here? And why not return?

Reply:
Oh, forgot about `return`.

Reply:
1b82e85770b5a4165caac68d265d83a2b58a8702